application/client.js

```diff
@@ -152,7 +152,14 @@ module.exports = HttpGateway.extend({
                     try {
                         responseData = JSON.parse(responseText);
                     } catch (e) {
-                        responseData = responseText;
```

Comment:
I guess what I mean to suggest is -- it may not be `http` causing that behavior. It might just be the underlying XHR implementation.